control/plugin/metric.go

```diff
@@ -90,8 +90,8 @@ func NewPluginConfigType() ConfigType {
 // Represents a metric type. Only used within plugins and across plugin calls.
 // Converted to core.MetricType before being used within modules.
 type MetricType struct {
-	// Namespace is the identifier for a metric.
-	Namespace_ []core.NamespaceElement `json:"namespace"`
+	// Namespace_ is the identifier for a metric.
+	Namespace_ core.Namespace `json:"namespace"`
```

Comment:
Hopefully not. But it could require massive changes. We'll need to find it out if it works with GRPC first.